datalake/src/main/java/com/sequenceiq/datalake/service/sdx/dr/SdxBackupRestoreService.java

```diff
@@ -557,34 +557,43 @@ private AttemptResult<DatalakeBackupStatusResponse> sdxFullDrSucceeded(SdxCluste
      * @param entitlementEnabled Whether the entitlement required for backups with this operation is enabled.
      * @return true if backup can be performed, False otherwise.
      */
+    @SuppressWarnings("checkstyle:NPathComplexity")
     public boolean shouldSdxBackupBePerformed(SdxCluster cluster, boolean entitlementEnabled) {
         String reason = null;
         if (!entitlementEnabled) {
-            reason = "Required entitlement for backup during this operation not enabled for this account.";
-        } else if (!datalakeDrConfig.isConfigured()) {
-            reason = "Datalake DR is not configured!";
-        } else {
-            DetailedEnvironmentResponse environmentResponse = environmentClientService.getByName(cluster.getEnvName());
-            CloudPlatform cloudPlatform = CloudPlatform.valueOf(environmentResponse.getCloudPlatform());
-            if (CloudPlatform.MOCK.equalsIgnoreCase(cloudPlatform.name())) {
-                return true;
-            }
+            LOGGER.info("Backup not triggered. Reason: Required entitlement for backup during this operation not enabled for this account");
+            return false;
+        }
+        if (!datalakeDrConfig.isConfigured()) {
+            LOGGER.info("Backup not triggered. Reason: Datalake DR is not configured!");
+            return false;
+        }
 
-            if (isVersionOlderThan(cluster, "7.2.1")) {
-                reason = "Unsupported runtime: " + cluster.getRuntime();
-            } else if (cluster.getCloudStorageFileSystemType() == null) {
-                reason = "Cloud storage not initialized";
-            } else if (cluster.getCloudStorageFileSystemType().isGcs()) {
-                reason = "Unsupported cloud provider GCS ";
-            } else if (cluster.getCloudStorageFileSystemType().isAdlsGen2() &&
-                    isVersionOlderThan(cluster, "7.2.2")) {
-                reason = "Unsupported cloud provider Azure on runtime: " + cluster.getRuntime();
-            }
+        DetailedEnvironmentResponse environmentResponse = environmentClientService.getByName(cluster.getEnvName());
+        CloudPlatform cloudPlatform = CloudPlatform.valueOf(environmentResponse.getCloudPlatform());
+        if (CloudPlatform.MOCK.equalsIgnoreCase(cloudPlatform.name())) {
+            LOGGER.info("Backup triggered on MOCK cloud platform");
+            return true;
         }
-        if (reason != null) {
-            LOGGER.info("Backup not triggered. Reason: " + reason);
+
+        if (isVersionOlderThan(cluster, "7.2.1")) {
```

Comment:
I don't think we'll ever trigger tests with 7.2.1, this condition could be removed.

Reply:
My mistake I though this is test code. Nevertheless I'll check wether this can be removed or not.